lib/galaxy_test/api/test_history_contents.py

```diff
@@ -738,31 +742,37 @@ def test_index_filter_by_type(self):
         contents_response = self._get(f"histories/{history_id}/contents?types=dataset_collection").json()
         assert len(contents_response) == expected_num_collections
 
-    def test_homogeneous_datatype_field(self):
+    def test_elements_datatypes_field(self):
         history_id = self.dataset_populator.new_history()
-        # List with all elements of txt datatype (homogeneous)
-        homogeneous_collection_name = "homogeneous"
-        create_homogeneous_response = self.dataset_collection_populator.upload_collection(history_id, "list", elements=[
-            {"name": "test0", "src": "pasted", "paste_content": "abc", "ext": "txt"},
-            {"name": "test1", "src": "pasted", "paste_content": "cba", "ext": "txt"},
-        ], name=homogeneous_collection_name)
+        collection_name = "homogeneous"
+        expected_datatypes = ["txt"]
+        elements = [  # List with all elements of txt datatype (homogeneous)
+            {"name": "test1", "src": "pasted", "paste_content": "abc", "ext": "txt"},
+            {"name": "test2", "src": "pasted", "paste_content": "abc", "ext": "txt"},
+        ]
+        self._upload_collection_list_with_elements(history_id, collection_name, elements)
+        self._assert_collection_has_expected_elements_datatypes(history_id, collection_name, expected_datatypes)
+
+        collection_name = "heterogeneous"
+        expected_datatypes = ["txt", "tabular"]
+        elements = [  # List with txt and tabular datatype (heterogeneous)
+            {"name": "test2", "src": "pasted", "paste_content": "abc", "ext": "txt"},
+            {"name": "test3", "src": "pasted", "paste_content": "a,b,c\n", "ext": "tabular"},
+        ]
```

Comment:
While updating the test I found that if I use the same element/dataset name the extensions are not correctly registered in the `dataset_dbkeys_and_extensions_summary` property, is this intended or did I misunderstand how it should work?
For example, changing both names to `test` above:
```diff
elements = [  # List with txt and tabular datatype (heterogeneous)
-    {"name": "test2", "src": "pasted", "paste_content": "abc", "ext": "txt"},
+    {"name": "test", "src": "pasted", "paste_content": "abc", "ext": "txt"},
-    {"name": "test3", "src": "pasted", "paste_content": "a,b,c\n", "ext": "tabular"},
+    {"name": "test", "src": "pasted", "paste_content": "a,b,c\n", "ext": "tabular"},
]
```
will return `{ "tabular" }` in hdca.dataset_dbkeys_and_extensions_summary[1] but I would expect it to be `{ "txt", "tabular" }`

Reply:
You can't have an element of the same name, we should error out when creating a collection like that though.

Reply:
Oh, I see, I thought it was more like the dataset name and not an identifier. Thank you, it makes sense now.